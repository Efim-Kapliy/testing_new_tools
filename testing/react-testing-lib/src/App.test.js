// import { render, screen, fireEvent } from '@testing-library/react';
// // Более полно эмулирует поведение пользователя, чем
// import userEvent from '@testing-library/user-event';
// import App from './App';

// describe('Test app', () => {
//   test('renders learn react link', () => {
//     render(<App />);
//     const helloWorldElem = screen.getByText(/hello world/i);
//     const btn = screen.getByRole('button');
//     const input = screen.getByPlaceholderText(/input value/i);
//     //Проверяем, что элемент появился в нашем DOM-дереве
//     expect(helloWorldElem).toBeInTheDocument();
//     expect(btn).toBeInTheDocument();
//     expect(input).toBeInTheDocument();
//     expect(input).toMatchSnapshot();
//   });

//   test('RENDERS SECOND', async () => {
//     render(<App />);
//     // const helloWorldElem = screen.queryByText(/hello world2/i);
//     // expect(helloWorldElem).toBeNull();
//     const helloWorldElem = await screen.findByText(/data/i);
//     expect(helloWorldElem).toBeInTheDocument();
//     // screen.debug();
//     expect(helloWorldElem).toHaveStyle({ color: 'red' });
//   });

//   test('CLICK EVENT', () => {
//     render(<App />);
//     const btn = screen.getByTestId('toggle-btn');
//     expect(screen.queryByTestId('toggle-elem')).toBeNull();
//     fireEvent.click(btn);
//     expect(screen.getByTestId('toggle-elem')).toBeInTheDocument();
//     fireEvent.click(btn);
//     expect(screen.queryByTestId('toggle-elem')).toBeNull();
//   });

//   test('INPUT EVENT', async () => {
//     const event = userEvent.setup();

//     render(<App />);
//     const input = screen.getByPlaceholderText(/input value/i);
//     expect(screen.queryByTestId('value-elem')).toContainHTML('');
//     // Искусственное событие
//     // fireEvent.input(input, {
//     //   target: { value: '369' },
//     // });
//     // Также искусственное событие
//     await event.type(input, '369');
//     expect(screen.getByTestId('value-elem')).toContainHTML('369');
//   });
// });
