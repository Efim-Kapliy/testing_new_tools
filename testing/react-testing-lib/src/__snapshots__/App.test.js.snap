// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`Test app renders learn react link 2`] = `
<input
  placeholder="input value"
  type="text"
/>
`;

exports[`renders learn react link 1`] = `
<input
  placeholder="input value"
  type="text"
/>
`;
